src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -955,34 +998,163 @@ void GoogleCalendarSyncAdaptor::eventsFinishedHandler()
+            } break;
+            case GoogleCalendarSyncAdaptor::Modify: {
+                mKCal::Notebook::Ptr notebook = notebookForCalendarId(accountId, calendarId);
+                if (notebook.isNull()) {
+                    SOCIALD_LOG_ERROR("unable to modify non-existent calendar:" << calendarId << "for account:" << accountId);
+                    m_syncSucceeded[accountId] = false; // we don't return immediately, as we want to at least attempt to
+                                                        // apply other database modifications if possible, in order to leave
+                                                        // the local database in a usable state even after failed sync.
+                } else {
+                    notebook->setIsReadOnly(false);
+                    notebook->setName(m_serverCalendarIdToCalendarInfo[accountId].value(calendarId).summary);
+                    notebook->setColor(m_serverCalendarIdToCalendarInfo[accountId].value(calendarId).color);
+                    notebook->setDescription(m_serverCalendarIdToCalendarInfo[accountId].value(calendarId).description);
+                    m_storage->updateNotebook(notebook);
+                    m_storageNeedsSave = true;
+                }
+            } break;
+            case GoogleCalendarSyncAdaptor::Delete: {
+                mKCal::Notebook::Ptr notebook = notebookForCalendarId(accountId, calendarId);
+                if (notebook.isNull()) {
+                    SOCIALD_LOG_ERROR("unable to delete non-existent calendar:" << calendarId << "for account:" << accountId);
+                    // m_syncSucceeded[accountId] = false; // don't mark as failed, since the outcome is identical.
+                } else {
+                    m_storage->loadNotebookIncidences(notebook->uid());
+                    KCalCore::Incidence::List incidenceList;
+                    m_storage->allIncidences(&incidenceList, notebook->uid());
+                    foreach (KCalCore::Incidence::Ptr incidence, incidenceList) {
+                        m_calendar->deleteIncidence(m_calendar->incidence(incidence->uid(), incidence->recurrenceId()));
+                    }
+                    m_storage->deleteNotebook(notebook);
+                    m_storageNeedsSave = true;
+                }
+            } break;
+            case GoogleCalendarSyncAdaptor::CleanSync: {
+                // delete
+                mKCal::Notebook::Ptr notebook = notebookForCalendarId(accountId, calendarId);
+                if (!notebook.isNull()) {
+                    m_storage->loadNotebookIncidences(notebook->uid());
+                    KCalCore::Incidence::List incidenceList;
+                    m_storage->allIncidences(&incidenceList, notebook->uid());
+                    foreach (KCalCore::Incidence::Ptr incidence, incidenceList) {
+                        m_calendar->deleteIncidence(m_calendar->incidence(incidence->uid(), incidence->recurrenceId()));
+                    }
```

Comment:
Let me test again, I may have been doing something strange when I saw the ghost events appear, or misattributed the cause.